File: ABC/ABC341/C/main.cc
#include <iostream>
#include <vector>

using namespace std;

int main() {
  int h, w, n;
  string t;
  cin >> h >> w >> n >> t;
  vector<string> s(h);
  for (auto &e : s) cin >> e;

  auto Func = [&](int r, int c) -> int {
    for (auto d : t) {
      if (d == 'L') {
        --c;
      } else if (d == 'R') {
        ++c;
      } else if (d == 'U') {
        --r;
      } else {
        ++r;
      }
      if (s[r][c] == '#') return 0;
    }
    return 1;
  };

  int res = 0;
  for (int i = 1; i < h - 1; ++i) {
    for (int j = 1; j < w - 1; ++j) {
      if (s[i][j] == '#') continue;
      res += Func(i, j);
    }
  }

  cout << res << endl;
  return 0;
}

File: ABC/ABC341/C/main.py
h, w, n = map(int, input().split())
t = input()
s = [input() for _ in range(h)]

mv = []
for d in t:
    if d == 'L':
        mv.append((0, -1))
    elif d == 'R':
        mv.append((0, 1))
    elif d == 'U':
        mv.append((-1, 0))
    else:
        mv.append((1, 0))

res = 0
for i in range(1, h - 1):
    for j in range(1, w - 1):
        if s[i][j] == '#':
            continue
        r = i
        c = j
        for dr, dc in mv:
            r += dr
            c += dc
            if s[r][c] == '#':
                break

        if s[r][c] == '.':
            res += 1

print(res)
